core/src/main/java/co/cask/wrangler/executor/TextDirectives.java

```diff
@@ -392,6 +394,51 @@ public TextDirectives(List<String> directives) {
         }
         break;
 
+        // parse-as-avro <column> <schema-id> <json|binary> [version]
+        case "parse-as-avro" : {
+          String column = getNextToken(tokenizer, command, "column", lineno);
+          String schemaId = getNextToken(tokenizer, command, "schema-id", lineno);
+          String type = getNextToken(tokenizer, command, "type", lineno);
+          if (!"json".equalsIgnoreCase(type) && !"binary".equalsIgnoreCase(type)) {
+           throw new DirectiveParseException(
+             String.format("Parsing AVRO can be either of type 'json' or 'binary'")
+           );
+          }
+          String versionOpt = getNextToken(tokenizer, "\n", command, "depth", lineno, true);
```

Comment:
Also, maybe there should be "getNextInt", "getNextDouble", etc. so that each directive doesn't have to do the parsing.